schemas.ts

```diff
@@ -0,0 +1,172 @@
+import _ from 'lodash';
+
+/**
+ * Gets the resource schema object for a resource
+ *
+ * @param resource
+ * @param resourceName
+ * @returns The resource schema
+ */
+const getResourceSchema = (resource: any, resourceName: string): any => {
+  const resourceSchema: any = {
```

Comment:
If we use the `object` type here, we're saying all the properties in its declaration are the only properties that the object can have. This won't work, since we add properties to it later. We can either give it an `any` type or declare a class or interface for it. I'm planning on looking into the latter in the future.

Reply:
Interesting. Sounds like a good plan.